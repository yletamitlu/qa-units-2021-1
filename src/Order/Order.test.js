jest.mock('../utils/getDate');

/* eslint-disable import/first */
import React from 'react'
import Order from "./Order";
import {configure, shallow, render} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {getDate} from "../utils/getDate";

configure({ adapter: new Adapter() });

describe('Order.js', () => {
    beforeEach(() => {
        getDate.mockReturnValue('27 11 1998')
    });

    it('should render a Order', () => {
        const order = {shop: 'shop', date: 'date'}
        const component = shallow(<Order order={order}/>);

        expect(component).toMatchSnapshot();
    });

    it('Order div counter', () => {
        const order = {
            shop: 'shop',
            date: 'date',
            "items": Array ['item'],}
        const component = render(<Order order={order}/>);

        expect(component.find('div')).toHaveLength(2);
        expect(component).toMatchSnapshot();
    });

    it('Order with empty date', () => {
        const order = {shop: 'shop', date: null};
        const component = shallow(<Order order={order}/>);

        expect(component.find('div')).toHaveLength(0);
        expect(component).toMatchSnapshot();
    });

    it('Order with empty order', () => {
        const order = null;
        const component = shallow(<Order order={order}/>);

        expect(component.find('div')).toHaveLength(0);
        expect(component).toMatchSnapshot();
    });

    it('Order with items', () => {
        const order = {shop: 'shop', date: 100500, items: ["some data"]}
        const component = shallow(<Order order={order}/>);

        expect(component.find('div')).toHaveLength(4);
        expect(component).toMatchSnapshot();
    });
});
