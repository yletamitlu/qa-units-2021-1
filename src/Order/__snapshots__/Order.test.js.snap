// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js Order div counter 1`] = `
<div
  class="Order"
>
  <div
    class="Order-header"
  >
    <span
      class="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    >
      27 11 1998
    </span>
  </div>
  <div
    class="Order-items"
  />
</div>
`;

exports[`Order.js Order with empty date 1`] = `""`;

exports[`Order.js Order with empty order 1`] = `""`;

exports[`Order.js Order with items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    >
      27 11 1998
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      some data
    </div>
  </div>
</div>
`;

exports[`Order.js should render a Order 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    >
      27 11 1998
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;
